iothub_client/inc/iothub_device_client.h

```diff
@@ -280,8 +275,7 @@ extern "C"
     * @param    userContextCallback         User specified context that will be provided to the
```

Comment:
* @brief This API sends a report of the device's properties and their current values. The device reported properties will be sent on the next call to `DoWork()`.

Reply:
If we want this it feels like a bigger change -- since almost every LL API would need something about work being queued.  Which in theory we've already covered in `threading_notes.md`.

I'm going to leave as-is for now and we can revisit if we think we need per-API clarity.